utils/s3/client.cc

```diff
@@ -284,14 +283,17 @@ future<> client::do_retryable_request(group_client& gc, http::request req, http:
     }
 }
 
-http::experimental::client::reply_handler client::make_s3_error_handler(http::experimental::client::reply_handler&& handler,
-                                                                        multipart_upload_completion is_mpu_completion) {
-    return [is_mpu_completion, handler{std::move(handler)}](const http::reply& rep, input_stream<char>&& in) -> future<> {
+http::experimental::client::reply_handler client::make_s3_error_handler(
+        http::experimental::client::reply_handler&& handler, http::reply::status_type expected, multipart_upload_completion is_mpu_completion) {
+    return [is_mpu_completion, handler{std::move(handler)}, expected](const http::reply& rep, input_stream<char>&& in) -> future<> {
         auto payload = std::move(in);
         auto error = co_await look_for_error(rep, payload, is_mpu_completion);
         if (error.get_error_type() != aws::aws_error_type::OK) {
             co_await coroutine::return_exception(aws::aws_exception(std::move(error)));
         }
+        if (rep._status != expected) {
```

Comment:
Do I get it right that prior to this patch http reply status was effectively ignored? Since (quoting the commit message)

> since the http::client::make_request call is done with `nullopt`

?

Reply:
yes. feels like this part should be a part of the previous commit, will fix